tests-integration/tests/grpc.rs

```diff
@@ -627,3 +628,64 @@ pub async fn test_prom_gateway_query(store_type: StorageType) {
     let _ = fe_grpc_server.shutdown().await;
     guard.remove_all().await;
 }
+
+pub async fn test_grpc_timezone(store_type: StorageType) {
+    let config = GrpcServerConfig {
+        max_recv_message_size: 1024,
+        max_send_message_size: 1024,
+    };
+    let (addr, mut guard, fe_grpc_server) =
+        setup_grpc_server_with(store_type, "auto_create_table", None, Some(config)).await;
+
+    let grpc_client = Client::with_urls(vec![addr]);
+    let mut db = Database::new_with_dbname(
+        format!("{}-{}", DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME),
+        grpc_client,
+    );
+    db.set_timezone("Asia/Shanghai");
+    let sys1 = to_batch(db.sql("show variables system_time_zone;").await.unwrap()).await;
+    let user1 = to_batch(db.sql("show variables time_zone;").await.unwrap()).await;
+    db.set_timezone("");
+    let sys2 = to_batch(db.sql("show variables system_time_zone;").await.unwrap()).await;
+    let user2 = to_batch(db.sql("show variables time_zone;").await.unwrap()).await;
+    assert_eq!(sys1, sys2);
+    assert_eq!(
+        sys2,
+        "\
++------------------+
+| SYSTEM_TIME_ZONE |
++------------------+
+| UTC              |
++------------------+"
+    );
+    assert_eq!(
+        user1,
+        "\
++---------------+
+| TIME_ZONE     |
++---------------+
+| Asia/Shanghai |
++---------------+"
+    );
+    assert_eq!(
+        user2,
+        "\
++-----------+
+| TIME_ZONE |
++-----------+
+| UTC       |
++-----------+"
+    );
+    let _ = fe_grpc_server.shutdown().await;
```

Comment:
I think we should add some tests that run SQL to retrieve timestamps and see if the timezone setting works.

Reply:
This pr is mainly about the ability to set timezone of user currently in. The current modification will not have any impact on the query results, so no related tests have been added. In the current PR, users can only set the time zone, and the time zone will not affect the behavior of any query.

Reply:
all right